File: timeish-app/src/app/models/time-sheet.ts
import { Activity } from "./activity";
import { PayPeriod } from "./pay-period";
import { Employee } from "./employee";

export class TimeSheet {
    id: number;
    issued: Date;
    submitted: Date;
    approved: Date;
    payPeriodId: number;
    payPeriod: PayPeriod;
    employeeId: number;
    employee: Employee;
    activities: Array<Activity>;

    constructor(instance) {
        this.activities = new Array<Activity>();   // need this for activities.length check
        Object.assign(this, instance);
        Object.assign(this.employee, new Employee(instance.employee))
    }

    totalHours(): number {
        let hours: number = 0;
        this.activities.forEach(activity => {
            hours += isNaN(activity.hours) ? 0 : activity.hours;
        });
        return hours;
    }

    totalPay(): number {
        let pay: number = 0;
        this.activities.forEach(activity => {
            pay += isNaN(activity.pay) ? 0 : activity.pay;
        })
        return pay;
    }
}

File: Tymish.Core/Entities/TimeSheet.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tymish.Core.Entities
{
    public class TimeSheet : BaseEntity
    {
        public DateTime Issued { get; set; }
        public DateTime? Submitted { get; set; }
        public DateTime? Approved { get; set; }
        [Required] public int EmployeeId { get; set; }
        public int? PayPeriodId { get; set; }

        public virtual Employee Employee{ get; set; }
        public virtual PayPeriod PayPeriod { get; set; }
        public virtual IEnumerable<Activity> Activities { get; set;}
    }

    public class Activity : BaseEntity
    {
        [Required] public DateTime Date { get; set; }
        [Required] [Range(0,24)] public int Hours { get; set; }
        public string Description { get; set; }
        [Required] public decimal Pay { get; set; }
        [Required] public int TimeSheetId { get; set; }
    }

    public class PayPeriod : BaseEntity
    {
        public DateTime Start { get; set; }
        public DateTime End { get; set; }
    }
}